### PR TITLE
docs: Add documentation and TODOs for cleanup

### PR DESCRIPTION
**Pull request by atscott**
* Document `currentUrlTree`, `browserUrlTree`, and `rawUrlTree`
* Add a couple `TODO` comments to investigate changes based on understanding

### CODE REVIEW

File: packages/router/src/router.ts [hunk 1]
```diff
@@ -397,8 +397,42 @@ export const subsetMatchOptions: IsActiveMatchOptions = {
...
+   * Represents the activated `UrlTree` that the `Router` is configured to handle (through
+   * `UrlHandlingStrategy`). That is, after we find the route config tree that we're going to
+   * activate, run guards, and are just about to activate the route, we set the currentUrlTree.
+   *
+   * This should match the `browserUrlTree` when a navigation succeeds. If the
+   * `UrlHandlingStrategy.shouldProcessUrl` is `false`, only the `browserUrlTree` is updated.
+   */
   private currentUrlTree: UrlTree;
+  /**
+   * Meant to represent the entire browser url after a successful navigataion. In the life of a
```
Comment (dario-piotrowicz) [inline packages/router/src/router.ts:410]:
```suggestion
   * Meant to represent the entire browser url after a successful navigation. In the life of a
```

File: packages/router/src/router.ts [hunk 2]
```diff
@@ -633,13 +667,22 @@ export class Router {
                      switchMap(t => {
                        const urlTransition = !this.navigated ||
                            t.extractedUrl.toString() !== this.browserUrlTree.toString();
+                       /* || this.browserUrlTree.toString() !== this.currentUrlTree.toString() */
+                       // TODO(atscott): Run TGP to see if the above change can be made. There are
+                       // situations where a navigation is cancel _after_ browserUrlTree is
```
Comment (mauromattos00) [inline packages/router/src/router.ts:672]:
```suggestion
                       // situations where a navigation is canceled _after_ browserUrlTree is
```

File: packages/router/src/router.ts [hunk 3]
```diff
@@ -855,7 +905,7 @@ export class Router {
                      tap((t: NavigationTransition) => {
                        this.currentUrlTree = t.urlAfterRedirects;
                        this.rawUrlTree =
-                           this.urlHandlingStrategy.merge(this.currentUrlTree, t.rawUrl);
+                           this.urlHandlingStrategy.merge(t.urlAfterRedirects, t.rawUrl);
```
Comment (AndrewKushnir) [inline packages/router/src/router.ts:908]: Just wanted to check if this change is intentional. Should we move that to a separate PR/commit? Would this change require tests?

**Review by AndrewKushnir — APPROVED** [review]
LGTM 👍 Just one quick question to make sure that the change was intentional.

File: packages/router/src/router.ts (hunk 3 above)
Comment (atscott) [reply packages/router/src/router.ts:908]: This change doesn't affect anything but makes the code more clear. If you look at the line about, `currentUrlTree` is set to `urlAfterRedirects`.

Comment (AndrewKushnir) [reply packages/router/src/router.ts:908]: Ah, I've missed that 🤦  Thanks Andrew.
